File: components/cv_screening.py
# components/cv_screening.py
import streamlit as st
from huggingface_hub import InferenceClient
from utils.helpers import extract_skills, get_match_score, get_recommendation
import os

def show_cv_screening():
    st.markdown("""
    <div style='padding-bottom: 2rem;'>
        <div style='display: flex; align-items: center; gap: 16px; margin-bottom: 0.75rem;'>
            <div style='
                width: 48px;
                height: 48px;
                background: linear-gradient(135deg, rgba(46,134,171,0.2) 0%, rgba(46,134,171,0.1) 100%);
                border: 1px solid rgba(46,134,171,0.3);
                border-radius: 12px;
                display: flex;
                align-items: center;
                justify-content: center;
            '>
                <span style='color: #2E86AB; font-size: 1.4rem;'>◆</span>
            </div>
            <div>
                <h1 style='font-size: 2rem; font-weight: 700; margin: 0; letter-spacing: -0.5px;'>CV Screening</h1>
                <p style='color: #8B949E; font-size: 0.95rem; margin: 4px 0 0;'>Analyze candidate qualifications against job requirements</p>
            </div>
        </div>
    </div>
    """, unsafe_allow_html=True)
    
    client = InferenceClient(token=os.getenv("HUGGINGFACE_TOKEN"))
    
    col1, col2 = st.columns(2)
    
    with col1:
        st.markdown("""
        <div style='
            background: linear-gradient(145deg, #161B22 0%, #1C2128 100%);
            border: 1px solid #30363D; 
            border-radius: 14px;
            padding: 1.5rem;
            margin-bottom: 0.75rem;
        '>
            <div style='display: flex; align-items: center; gap: 12px;'>
                <div style='
                    width: 36px;
                    height: 36px;
                    background: rgba(46, 134, 171, 0.15);
                    border: 1px solid rgba(46, 134, 171, 0.2);
                    border-radius: 8px;
                    display: flex;
                    align-items: center;
                    justify-content: center;
                '>
                    <span style='color: #2E86AB; font-size: 1rem;'>◆</span>
                </div>
                <div>
                    <h3 style='margin: 0; color: #FFFFFF; font-size: 1.05rem; font-weight: 600;'>Job Description</h3>
                    <p style='margin: 2px 0 0; color: #6E7681; font-size: 0.8rem;'>Enter the role requirements</p>
                </div>
            </div>
        </div>
        """, unsafe_allow_html=True)
        job_desc = st.text_area(
            "Paste job description here:",
            height=320,
            placeholder="e.g. We need a Python developer with 2 years experience in ML...",
            label_visibility="collapsed"
        )
    
    with col2:
        st.markdown("""
        <div style='
            background: linear-gradient(145deg, #161B22 0%, #1C2128 100%);
            border: 1px solid #30363D; 
            border-radius: 14px;
            padding: 1.5rem;
            margin-bottom: 0.75rem;
        '>
            <div style='display: flex; align-items: center; gap: 12px;'>
                <div style='
                    width: 36px;
                    height: 36px;
                    background: rgba(0, 212, 170, 0.15);
                    border: 1px solid rgba(0, 212, 170, 0.2);
                    border-radius: 8px;
                    display: flex;
                    align-items: center;
                    justify-content: center;
                '>
                    <span style='color: #00D4AA; font-size: 1rem;'>○</span>
                </div>
                <div>
                    <h3 style='margin: 0; color: #FFFFFF; font-size: 1.05rem; font-weight: 600;'>Candidate CV</h3>
                    <p style='margin: 2px 0 0; color: #6E7681; font-size: 0.8rem;'>Enter the candidate's resume</p>
                </div>
            </div>
        </div>
        """, unsafe_allow_html=True)
        cv_text = st.text_area(
            "Paste candidate CV here:",
            height=320,
            placeholder="e.g. John Doe, 3 years Python experience, worked on ML projects...",
            label_visibility="collapsed"
        )
    
    st.markdown("<div style='margin: 2rem 0;'></div>", unsafe_allow_html=True)
    
    col_center = st.columns([1, 2, 1])
    with col_center[1]:
        if st.button("Analyze Candidate", use_container_width=True):
            if job_desc and cv_text:
                with st.spinner("AI is analyzing the CV..."):
                    score, percentage = get_match_score(job_desc, cv_text)
                    job_skills = extract_skills(job_desc)
                    cv_skills = extract_skills(cv_text)
                    matched_skills = [s for s in job_skills if s in cv_skills]
                    missing_skills = [s for s in job_skills if s not in cv_skills]
                    recommendation = get_recommendation(percentage, matched_skills, missing_skills)

                    analysis_text = (
                        f"Job requires: {job_desc[:300]}. "
                        f"Candidate has: {cv_text[:300]}. "
                        f"Matched skills: {', '.join(matched_skills)}. "
                        f"Missing skills: {', '.join(missing_skills)}. "
                        f"Overall match score: {percentage}%."
                    )
                    ai_summary = client.summarization(
                        analysis_text,
                        model="facebook/bart-large-cnn"
                    )
                    explanation = ai_summary.summary_text

                st.markdown("<hr style='margin: 2.5rem 0;'>", unsafe_allow_html=True)
                
                st.markdown("""
                <div style='padding-bottom: 1.5rem;'>
                    <div style='display: flex; align-items: center; gap: 12px;'>
                        <div style='
                            width: 40px;
                            height: 40px;
                            background: rgba(0, 212, 170, 0.15);
                            border: 1px solid rgba(0, 212, 170, 0.2);
                            border-radius: 10px;
                            display: flex;
                            align-items: center;
                            justify-content: center;
                        '>
                            <span style='color: #00D4AA; font-size: 1.1rem;'>◆</span>
                        </div>
                        <div>
                            <h2 style='font-size: 1.5rem; font-weight: 700; margin: 0;'>Screening Results</h2>
                            <p style='color: #8B949E; font-size: 0.85rem; margin: 4px 0 0;'>AI-powered candidate analysis</p>
                        </div>
                    </div>
                </div>
                """, unsafe_allow_html=True)
                
                if percentage >= 70:
                    score_color = "#00D4AA"
                    score_label = "Strong Match"
                    score_bg = "rgba(0, 212, 170, 0.08)"
                    score_border = "rgba(0, 212, 170, 0.2)"
                elif percentage >= 40:
                    score_color = "#FFA500"
                    score_label = "Partial Match"
                    score_bg = "rgba(255, 165, 0, 0.08)"
                    score_border = "rgba(255, 165, 0, 0.2)"
                else:
                    score_color = "#FF4B4B"
                    score_label = "Weak Match"
                    score_bg = "rgba(255, 75, 75, 0.08)"
                    score_border = "rgba(255, 75, 75, 0.2)"
                
                st.markdown(f"""
                <div style='
                    background: {score_bg};
                    border: 1px solid {score_border};
                    border-radius: 16px;
                    padding: 2rem;
                    text-align: center;
                    margin-bottom: 1.5rem;
                    position: relative;
                    overflow: hidden;
                '>
                    <div style='
                        position: absolute;
                        top: -50%;
                        left: 50%;
                        transform: translateX(-50%);
                        width: 300px;
                        height: 300px;
                        background: radial-gradient(circle, {score_color}15 0%, transparent 60%);
                        pointer-events: none;
                    '></div>
                    <p style='color: #8B949E; font-size: 0.9rem; margin: 0 0 0.5rem 0; text-transform: uppercase; letter-spacing: 1px;'>Match Score</p>
                    <h1 style='
                        font-size: 5rem; 
                        font-weight: 800; 
                        margin: 0;
                        color: {score_color};
                        letter-spacing: -3px;
                        line-height: 1;
                    '>{percentage}%</h1>
                    <div style='
                        display: inline-block;
                        background: {score_color}20;
                        border: 1px solid {score_color}40;
                        border-radius: 20px;
                        padding: 0.5rem 1.25rem;
                        margin-top: 1rem;
                    '>
                        <span style='color: {score_color}; font-size: 0.95rem; font-weight: 600;'>{score_label}</span>
                    </div>
                </div>
                """, unsafe_allow_html=True)
                
                st.progress(score)
                
                st.markdown(f"""
                <div style='
                    background: linear-gradient(145deg, #161B22 0%, #1C2128 100%);
                    border: 1px solid #30363D;
                    border-radius: 14px;
                    padding: 1.75rem;
                    margin: 2rem 0;
                '>
                    <div style='display: flex; align-items: center; gap: 14px; margin-bottom: 1.25rem;'>
                        <div style='
                            width: 40px;
                            height: 40px;
                            background: rgba(46, 134, 171, 0.15);
                            border: 1px solid rgba(46, 134, 171, 0.2);
                            border-radius: 10px;
                            display: flex;
                            align-items: center;
                            justify-content: center;
                        '>
                            <span style='color: #2E86AB;'>◆</span>
                        </div>
                        <h3 style='margin: 0; color: #FFFFFF; font-size: 1.15rem; font-weight: 600;'>Hiring Recommendation</h3>
                    </div>
                    <p style='color: #8B949E; line-height: 1.7; margin: 0; font-size: 0.95rem;'>{recommendation}</p>
                </div>
                """, unsafe_allow_html=True)
                
                col1, col2 = st.columns(2)
                with col1:
                    st.markdown("""
                    <div style='
                        background: linear-gradient(145deg, #161B22 0%, #1C2128 100%);
                        border: 1px solid #30363D;
                        border-radius: 14px;
                        padding: 1.5rem;
                        height: 100%;
                    '>
                        <div style='display: flex; align-items: center; gap: 12px; margin-bottom: 1.25rem;'>
                            <span style='color: #00D4AA; font-size: 1.2rem;'>●</span>
                            <h4 style='margin: 0; color: #FFFFFF; font-size: 1rem; font-weight: 600;'>Matched Skills</h4>
                        </div>
                    """, unsafe_allow_html=True)
                    
                    if matched_skills:
                        for skill in matched_skills:
                            st.markdown(f"""
                            <span style='
                                display: inline-block;
                                background: rgba(0, 212, 170, 0.1);
                                border: 1px solid rgba(0, 212, 170, 0.25);
                                border-radius: 8px;
                                padding: 0.6rem 1.1rem;
                                margin: 0.35rem;
                                color: #00D4AA;
                                font-size: 0.9rem;
                                font-weight: 500;
                            '>{skill}</span>
                            """, unsafe_allow_html=True)
                    else:
                        st.markdown("<p style='color: #8B949E; font-size: 0.9rem;'>No matched skills found.</p>", unsafe_allow_html=True)
                    st.markdown("</div>", unsafe_allow_html=True)
                
                with col2:
                    st.markdown("""
                    <div style='
                        background: linear-gradient(145deg, #161B22 0%, #1C2128 100%);
                        border: 1px solid #30363D;
                        border-radius: 14px;
                        padding: 1.5rem;
                        height: 100%;
                    '>
                        <div style='display: flex; align-items: center; gap: 12px; margin-bottom: 1.25rem;'>
                            <span style='color: #FF4B4B; font-size: 1.2rem;'>●</span>
                            <h4 style='margin: 0; color: #FFFFFF; font-size: 1rem; font-weight: 600;'>Missing Skills</h4>
                        </div>
                    """, unsafe_allow_html=True)
                    
                    if missing_skills:
                        for skill in missing_skills:
                            st.markdown(f"""
                            <span style='
                                display: inline-block;
                                background: rgba(255, 75, 75, 0.1);
                                border: 1px solid rgba(255, 75, 75, 0.25);
                                border-radius: 8px;
                                padding: 0.6rem 1.1rem;
                                margin: 0.35rem;
                                color: #FF4B4B;
                                font-size: 0.9rem;
                                font-weight: 500;
                            '>{skill}</span>
                            """, unsafe_allow_html=True)
                    else:
                        st.markdown("""
                        <div style='
                            background: rgba(0, 212, 170, 0.1);
                            border: 1px solid rgba(0, 212, 170, 0.2);
                            border-radius: 8px;
                            padding: 1rem;
                            text-align: center;
                        '>
                            <span style='color: #00D4AA; font-size: 0.9rem; font-weight: 500;'>No missing skills — great candidate!</span>
                        </div>
                        """, unsafe_allow_html=True)
                    st.markdown("</div>", unsafe_allow_html=True)
                
                st.markdown(f"""
                <div style='
                    background: linear-gradient(145deg, #161B22 0%, #1C2128 100%);
                    border: 1px solid #30363D;
                    border-radius: 14px;
                    padding: 1.75rem;
                    margin-top: 2rem;
                '>
                    <div style='display: flex; align-items: center; gap: 14px; margin-bottom: 1.25rem;'>
                        <div style='
                            width: 44px;
                            height: 44px;
                            background: linear-gradient(135deg, #1E3A5F 0%, #2E86AB 100%);
                            border-radius: 10px;
                            display: flex;
                            align-items: center;
                            justify-content: center;
                        '>
                            <span style='color: white; font-weight: 700; font-size: 0.9rem;'>AI</span>
                        </div>
                        <div>
                            <h3 style='margin: 0; color: #FFFFFF; font-size: 1.15rem; font-weight: 600;'>AI Analysis</h3>
                            <p style='margin: 2px 0 0; color: #6E7681; font-size: 0.8rem;'>Detailed assessment</p>
                        </div>
                    </div>
                    <p style='color: #8B949E; line-height: 1.75; margin: 0; font-size: 0.95rem;'>{explanation}</p>
                </div>
                """, unsafe_allow_html=True)
                
            else:
                st.markdown("""
                <div style='
                    background: rgba(255, 165, 0, 0.1);
                    border: 1px solid rgba(255, 165, 0, 0.25);
                    border-radius: 12px;
                    padding: 1.25rem 1.5rem;
                '>
                    <div style='display: flex; align-items: center; gap: 12px;'>
                        <span style='color: #FFA500; font-size: 1.2rem;'>!</span>
                        <span style='color: #FFA500; font-weight: 500;'>Please paste both Job Description and CV to analyze.</span>
                    </div>
                </div>
                """, unsafe_allow_html=True)

File: frontend/pages/cv-screening.js
import React, { useState } from 'react';

const SKILLS_LIST = [
  "python", "sql", "java", "javascript", "typescript",
  "machine learning", "ml", "data science", "pandas", "numpy",
  "scikit-learn", "tensorflow", "keras", "pytorch", "deep learning",
  "nlp", "natural language processing", "react", "nodejs", "angular",
  "vue", "html", "css", "django", "flask", "fastapi",
  "excel", "powerpoint", "tableau", "power bi", "aws", "azure",
  "docker", "kubernetes", "git", "linux", "sql server",
  "mongodb", "postgresql", "mysql", "redis", "elasticsearch",
  "communication", "teamwork", "leadership", "problem solving",
  "management", "analysis", "project management", "agile", "scrum"
];

function extractSkills(text) {
  const found = [];
  const textLower = text.toLowerCase();
  for (const skill of SKILLS_LIST) {
    if (textLower.includes(skill)) {
      found.push(skill.charAt(0).toUpperCase() + skill.slice(1));
    }
  }
  return [...new Set(found)];
}

function tokenize(text) {
  return text.toLowerCase().split(/\W+/).filter(word => word.length > 1);
}

function termFrequency(term, tokens) {
  const count = tokens.filter(t => t === term).length;
  return count / tokens.length;
}

function inverseDocumentFrequency(term, documents) {
  const numDocsWithTerm = documents.filter(doc => doc.includes(term)).length;
  if (numDocsWithTerm === 0) return 0;
  return Math.log(documents.length / numDocsWithTerm);
}

function cosineSimilarity(vec1, vec2) {
  const allTerms = [...new Set([...Object.keys(vec1), ...Object.keys(vec2)])];
  const dotProduct = allTerms.reduce((sum, term) => sum + (vec1[term] || 0) * (vec2[term] || 0), 0);
  const mag1 = Math.sqrt(Object.values(vec1).reduce((sum, val) => sum + val * val, 0));
  const mag2 = Math.sqrt(Object.values(vec2).reduce((sum, val) => sum + val * val, 0));
  if (mag1 === 0 || mag2 === 0) return 0;
  return dotProduct / (mag1 * mag2);
}

function calculateTFIDF(text1, text2) {
  const tokens1 = tokenize(text1);
  const tokens2 = tokenize(text2);
  const documents = [text1.toLowerCase(), text2.toLowerCase()];
  const allTerms = [...new Set([...tokens1, ...tokens2])];
  const tfidf1 = {};
  const tfidf2 = {};
  for (const term of allTerms) {
    tfidf1[term] = termFrequency(term, tokens1) * inverseDocumentFrequency(term, documents);
    tfidf2[term] = termFrequency(term, tokens2) * inverseDocumentFrequency(term, documents);
  }
  return cosineSimilarity(tfidf1, tfidf2);
}

function calculateScore(jobDesc, cvText) {
  const jobSkills = extractSkills(jobDesc);
  const cvSkills = extractSkills(cvText);
  if (!jobSkills.length) {
    return { score: 0, percentage: 0 };
  }
  const matched = jobSkills.filter(s => 
    cvSkills.some(cs => cs.toLowerCase() === s.toLowerCase())
  );
  const matchedCount = matched.length;
  const totalRequired = jobSkills.length;
  const skillsPercentage = (matchedCount / totalRequired) * 100;
  const tfidfScore = calculateTFIDF(jobDesc, cvText) * 100;
  const percentage = Math.min(Math.round((skillsPercentage * 0.7 + tfidfScore * 0.3) * 10) / 10, 100);
  return {
    score: percentage / 100,
    percentage
  };
}

function getScoreClass(percentage) {
  if (percentage >= 70) {
    return { class: "high", label: "Strong Match", color: "#00D4AA" };
  } else if (percentage >= 40) {
    return { class: "medium", label: "Partial Match", color: "#FFA500" };
  }
  return { class: "low", label: "Weak Match", color: "#FF4B4B" };
}

function generateRecommendation(percentage, matchedSkills, missingSkills) {
  const matchedCount = matchedSkills?.length || 0;
  const missingCount = missingSkills?.length || 0;
  
  if (percentage >= 70) {
    return `STRONG HIRE - ${percentage}% Match. This candidate is an excellent fit with ${matchedCount} matched skills including ${matchedSkills?.slice(0, 3).join(', ') || 'N/A'}. Recommended for immediate technical interview.`;
  } else if (percentage >= 40) {
    return `MAYBE - ${percentage}% Match. This candidate partially meets requirements with ${matchedCount} matched skills: ${matchedSkills?.join(', ') || 'None'}. Missing ${missingCount} skills: ${missingSkills?.join(', ') || 'None'}. Consider for interview if no stronger candidates available.`;
  }
  return `REJECT - ${percentage}% Match. This candidate does not meet minimum requirements. Only matches ${matchedCount} skills. Missing ${missingCount} critical skills. Do not proceed with this candidate.`;
}

export default function CVScreening() {
  const [jobDesc, setJobDesc] = useState('');
  const [cvText, setCvText] = useState('');
  const [loading, setLoading] = useState(false);
  const [results, setResults] = useState(null);
  const [error, setError] = useState('');

  const handleSubmit = async () => {
    if (!jobDesc || !cvText) {
      setError('Please paste both Job Description and CV to analyze.');
      return;
    }
    
    setLoading(true);
    setError('');
    setResults(null);
    
    try {
      await new Promise(r => setTimeout(r, 500));
      
      const { score, percentage } = calculateScore(jobDesc, cvText);
      const jobSkills = extractSkills(jobDesc);
      const cvSkills = extractSkills(cvText);
      
      const matchedSkills = jobSkills.filter(s => 
        cvSkills.some(cs => cs.toLowerCase() === s.toLowerCase())
      );
      const missingSkills = jobSkills.filter(s => 
        !cvSkills.some(cs => cs.toLowerCase() === s.toLowerCase())
      );
      
      const recommendation = generateRecommendation(percentage, matchedSkills, missingSkills);
      const { class: scoreClass, label: scoreLabel, color: scoreColor } = getScoreClass(percentage);
      
      setResults({
        score,
        percentage,
        matched_skills: matchedSkills,
        missing_skills: missingSkills,
        recommendation,
        ai_analysis: `Analysis complete. Match score is ${percentage}%. ${recommendation}`,
        score_class: scoreClass,
        score_label: scoreLabel,
        score_color: scoreColor
      });
    } catch (err) {
      setError('An error occurred. Please try again.');
    } finally {
      setLoading(false);
    }
  };

  return (
    <div>
      <div className="page-header">
        <div className="page-header-wrapper">
          <div className="page-icon">
            <svg width="20" height="20" viewBox="0 0 24 24" fill="none" stroke="currentColor" strokeWidth="2">
              <path d="M14 2H6a2 2 0 0 0-2 2v16a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V8z"></path>
              <polyline points="14 2 14 8 20 8"></polyline>
            </svg>
          </div>
          <div>
            <h1 className="page-title">CV Screening</h1>
            <p className="page-subtitle">Analyze candidate qualifications against job requirements</p>
          </div>
        </div>
      </div>

      <div className="grid-2">
        <div className="card">
          <div className="card-header">
            <div className="card-icon secondary">
              <svg width="16" height="16" viewBox="0 0 24 24" fill="none" stroke="currentColor" strokeWidth="2">
                <path d="M14 2H6a2 2 0 0 0-2 2v16a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V8z"></path>
                <polyline points="14 2 14 8 20 8"></polyline>
              </svg>
            </div>
            <div>
              <h3 className="card-title">Job Description</h3>
              <p className="card-subtitle">Enter the role requirements</p>
            </div>
          </div>
          <textarea
            className="form-textarea"
            placeholder="e.g. We need a Python developer with 2 years experience in ML..."
            value={jobDesc}
            onChange={(e) => setJobDesc(e.target.value)}
            style={{ height: '200px' }}
          />
        </div>

        <div className="card">
          <div className="card-header">
            <div className="card-icon primary">
              <svg width="16" height="16" viewBox="0 0 24 24" fill="none" stroke="currentColor" strokeWidth="2">
                <path d="M20 21v-2a4 4 0 0 0-4-4H8a4 4 0 0 0-4 4v2"></path>
                <circle cx="12" cy="7" r="4"></circle>
              </svg>
            </div>
            <div>
              <h3 className="card-title">Candidate CV</h3>
              <p className="card-subtitle">Enter the candidate's resume</p>
            </div>
          </div>
          <textarea
            className="form-textarea"
            placeholder="e.g. John Doe, 3 years Python experience, worked on ML projects..."
            value={cvText}
            onChange={(e) => setCvText(e.target.value)}
            style={{ height: '200px' }}
          />
        </div>
      </div>

      <div style={{ textAlign: 'center', margin: '2rem 0' }}>
        <button className="btn btn-primary" onClick={handleSubmit} disabled={loading} style={{ maxWidth: '300px' }}>
          {loading ? 'Analyzing...' : 'Analyze Candidate'}
        </button>
      </div>

      {loading && (
        <div className="spinner">
          <div className="spinner-ring"></div>
        </div>
      )}

      {error && (
        <div className="alert warning">
          <span>{error}</span>
        </div>
      )}

      {results && (
        <div>
          <hr className="divider" />

          <div className={`score-card ${results.score_class}`}>
            <p style={{ color: 'var(--text-secondary)', fontSize: '0.9rem', textTransform: 'uppercase', letterSpacing: '1px' }}>
              Match Score
            </p>
            <h1 className={`score-value ${results.score_class}`}>{results.percentage}%</h1>
            <div className={`score-label ${results.score_class}`}>{results.score_label}</div>
          </div>

          <div className="card" style={{ marginBottom: '1.5rem' }}>
            <h3 style={{ marginBottom: '1rem' }}>Hiring Recommendation</h3>
            <p style={{ color: 'var(--text-secondary)', lineHeight: 1.7 }}>{results.recommendation}</p>
          </div>

          <div className="grid-2">
            <div className="card">
              <h4 className="card-title" style={{ marginBottom: '1rem' }}>
                <span style={{ color: 'var(--accent)', marginRight: '8px' }}>&#9679;</span>
                Matched Skills
              </h4>
              <div className="skills-container">
                {results.matched_skills && results.matched_skills.length > 0 ? (
                  results.matched_skills.map((skill, idx) => (
                    <span key={idx} className="skill-badge matched">{skill}</span>
                  ))
                ) : (
                  <p style={{ color: 'var(--text-secondary)' }}>No matched skills found.</p>
                )}
              </div>
            </div>

            <div className="card">
              <h4 className="card-title" style={{ marginBottom: '1rem' }}>
                <span style={{ color: 'var(--error)', marginRight: '8px' }}>&#9679;</span>
                Missing Skills
              </h4>
              <div className="skills-container">
                {results.missing_skills && results.missing_skills.length > 0 ? (
                  results.missing_skills.map((skill, idx) => (
                    <span key={idx} className="skill-badge missing">{skill}</span>
                  ))
                ) : (
                  <div className="alert success" style={{ margin: 0 }}>No missing skills!</div>
                )}
              </div>
            </div>
          </div>

          <div className="card" style={{ marginTop: '1.5rem' }}>
            <h3 className="card-title" style={{ marginBottom: '1rem' }}>AI Analysis</h3>
            <p style={{ color: 'var(--text-secondary)', lineHeight: 1.75 }}>{results.ai_analysis}</p>
          </div>
        </div>
      )}
    </div>
  );
}